dev/flake8_awkward.py

```diff
@@ -0,0 +1,57 @@
+import ast
+import sys
+from typing import NamedTuple, Iterator
+
+
+class Flake8ASTErrorInfo(NamedTuple):
+    line_number: int
+    offset: int
+    msg: str
+    cls: type  # unused
+
+
+class Visitor(ast.NodeVisitor):
+    msg = "AK101 exception must be wrapped in ak._v2._util.*error"
+
+    def __init__(self):
+        self.errors: list[Flake8ASTErrorInfo] = []
+
+    def visit_Raise(self, node):
+        if isinstance(node.exc, ast.Call):
+            if isinstance(node.exc.func, ast.Attribute):
+                if node.exc.func.attr in {"error", "indexerror"}:
+                    return
+            if node.exc.func.id in {"ImportError"}:
+                return
+
+        self.errors.append(
+            Flake8ASTErrorInfo(node.lineno, node.col_offset, self.msg, type(self))
+        )
+
+
+class AwkwardASTPlugin:
+    name = "flake8_awkward"
+    version = "0.0.0"
+
+    def __init__(self, tree: ast.AST) -> None:
+        self._tree = tree
+
+    def run(self) -> Iterator[Flake8ASTErrorInfo]:
+        visitor = Visitor()
+        visitor.visit(self._tree)
+        yield from visitor.errors
+
+
+def main(path):
+    with open(path) as f:
+        code = f.read()
+
+    node = ast.parse(code)
+    plugin = AwkwardASTPlugin(node)
+    for err in plugin.run():
+        print(f"{path}:{err.line_number}:{err.offset} {err.msg}")
+
+
+if __name__ == "__main__":
+    for item in sys.argv[1:]:
+        main(item)
```

Comment:
This part is only for debugging

Reply:
Okay, visitor pattern, checking each attribute individually—that all is fine. Better/more formal, in fact: I was trying to make it simple.

The `unparse` could only ever be applied to an expression that is called, which is hardly ever anything more than a Name with Attributes, and it was only doing that if it's to the right of a Raise, so I think the execution time was pretty well controlled.

But this is really nice! (And now I have an example I'll be looking up if I ever want to add another one.) Thanks!